src/main/java/com/aws/greengrass/util/platforms/windows/WindowsRunasProcess.java

```diff
@@ -0,0 +1,450 @@
+            }
+        }
+    }
+
+    @Override
+    public OutputStream getOutputStream() {
+        return stdin;
+    }
+
+    @Override
+    public InputStream getInputStream() {
+        return stdout;
+    }
+
+    @Override
+    public InputStream getErrorStream() {
+        return stderr;
+    }
+
+    @Override
+    public synchronized int waitFor() throws InterruptedException {
+        if (procInfo == null) {
+            throw new RuntimeException("process was not created");
+        }
+        if (exited) {
+            return exitCode;
+        }
+        if (WinBase.WAIT_FAILED == Kernel32.INSTANCE.WaitForSingleObject(procInfo.hProcess, Kernel32.INFINITE)) {
+            throw lastErrorRuntimeException();
+        }
+        return exitValue();
+    }
+
+    @Override
+    public synchronized int exitValue() {
+        if (procInfo == null) {
+            throw new RuntimeException("process was not created");
+        }
+        if (exited) {
+            return exitCode;
+        }
+        IntByReference exitCodeRef = new IntByReference();
+        if (!Kernel32.INSTANCE.GetExitCodeProcess(procInfo.hProcess, exitCodeRef)) {
+            throw lastErrorRuntimeException();
+        }
+        if (WinBase.STILL_ACTIVE == exitCodeRef.getValue()) {
+            throw new IllegalThreadStateException("process hasn't exited");
+        }
+        exited = true;
+        exitCode = exitCodeRef.getValue();
+        return exitCode;
+    }
+
+    @Override
+    public void destroy() {
+        destroyForcibly();
+    }
+
+    @Override
+    public synchronized Process destroyForcibly() {
+        if (procInfo == null || exited) {
+            return this;
+        }
+        Kernel32.INSTANCE.TerminateProcess(procInfo.hProcess, EXIT_CODE_TERMINATED);
+        closeHandles();
+        exited = true;
+        exitCode = EXIT_CODE_TERMINATED;
+        return this;
+    }
+
+    private static LastErrorException lastErrorRuntimeException() {
+        return new LastErrorException(Kernel32.INSTANCE.GetLastError());
+    }
+
+    private static ProcessCreationException lastErrorProcessCreationException(String context) {
+        return new ProcessCreationException(String.format("[%s] %s", context, Kernel32Util.getLastErrorMessage()));
+    }
+
+    private static boolean isService(WinNT.HANDLEByReference processToken) throws ProcessCreationException {
+        // Get required buffer size
+        IntByReference returnLength = new IntByReference();
+        if (!Advapi32.INSTANCE.GetTokenInformation(processToken.getValue(), WinNT.TOKEN_INFORMATION_CLASS.TokenGroups,
+                null, 0, returnLength)) {
+            int lastErrorCode = Kernel32.INSTANCE.GetLastError();
+            if (lastErrorCode != WinError.ERROR_INSUFFICIENT_BUFFER) {
+                throw new ProcessCreationException(String.format("[%s] %s", "GetTokenInformation",
+                        Kernel32Util.formatMessageFromLastErrorCode(lastErrorCode)));
```

Comment:
throw `lastErrorProcessCreationException("GetTokenInformation")`

Reply:
Problem here is that we already called GetLastError above this line, calling it again will give no error. Did it this way because here we want to look at the actual error code and determine what to do.